utilities/rpp-unittests/HOST_NEW/Tensor_host_pkd3.cpp

```diff
@@ -401,39 +528,94 @@ int main(int argc, char **argv)
+    }
+    else
+    {
+        dstDescPtr->c = ip_channel;
+    }
 
     // Optionally set w stride as a multiple of 8 for src/dst
 
     srcDescPtr->w = ((srcDescPtr->w / 8) * 8) + 8;
     dstDescPtr->w = ((dstDescPtr->w / 8) * 8) + 8;
 
     // Set n/c/h/w strides for src/dst
-
-    srcDescPtr->strides.nStride = srcDescPtr->c * srcDescPtr->w * srcDescPtr->h;
-    srcDescPtr->strides.hStride = srcDescPtr->c * srcDescPtr->w;
-    srcDescPtr->strides.wStride = srcDescPtr->c;
-    srcDescPtr->strides.cStride = 1;
+    if (layout_switch == 0)
+    {
+        srcDescPtr->strides.nStride = srcDescPtr->c * srcDescPtr->w * srcDescPtr->h;
+        srcDescPtr->strides.hStride = srcDescPtr->c * srcDescPtr->w;
+        srcDescPtr->strides.wStride = srcDescPtr->c;
+        srcDescPtr->strides.cStride = 1;
+    }
+    if (layout_switch == 1)
+    {
+        srcDescPtr->strides.nStride = srcDescPtr->c * srcDescPtr->w * srcDescPtr->h;
+        srcDescPtr->strides.cStride = srcDescPtr->w * srcDescPtr->h;
+        srcDescPtr->strides.hStride = srcDescPtr->w;
+        srcDescPtr->strides.wStride = 1;
+    }
+    if (layout_switch == 2)
+    {
+        srcDescPtr->strides.nStride = ip_channel * srcDescPtr->w * srcDescPtr->h;
+        srcDescPtr->strides.cStride = srcDescPtr->w * srcDescPtr->h;
+        srcDescPtr->strides.hStride = srcDescPtr->w;
+        srcDescPtr->strides.wStride = 1;
+    }
 
     if (dstDescPtr->layout == RpptLayout::NHWC)
     {
-        dstDescPtr->strides.nStride = dstDescPtr->c * dstDescPtr->w * dstDescPtr->h;
-        dstDescPtr->strides.hStride = dstDescPtr->c * dstDescPtr->w;
-        dstDescPtr->strides.wStride = dstDescPtr->c;
-        dstDescPtr->strides.cStride = 1;
+        if (layout_switch == 0)
+        {
+            srcDescPtr->strides.nStride = srcDescPtr->c * srcDescPtr->w * srcDescPtr->h;
+            srcDescPtr->strides.hStride = srcDescPtr->c * srcDescPtr->w;
+            srcDescPtr->strides.wStride = srcDescPtr->c;
+            srcDescPtr->strides.cStride = 1;
+        }
+        if (layout_switch == 1)
+        {
+            dstDescPtr->strides.nStride = dstDescPtr->c * dstDescPtr->w * dstDescPtr->h;
+            dstDescPtr->strides.hStride = dstDescPtr->c * dstDescPtr->w;
+            dstDescPtr->strides.wStride = dstDescPtr->c;
+            dstDescPtr->strides.cStride = 1;
+        }
+        if (layout_switch == 2)
+        {
+            dstDescPtr->strides.nStride = ip_channel * dstDescPtr->w * dstDescPtr->h;
+            dstDescPtr->strides.hStride = ip_channel * dstDescPtr->w;
+            dstDescPtr->strides.wStride = ip_channel;
+            dstDescPtr->strides.cStride = 1;
+        }
     }
     else if (dstDescPtr->layout == RpptLayout::NCHW)
     {
-        dstDescPtr->strides.nStride = dstDescPtr->c * dstDescPtr->w * dstDescPtr->h;
+        if (layout_switch == 2)
+        {
+            dstDescPtr->strides.nStride = ip_channel * dstDescPtr->w * dstDescPtr->h;
```

Comment:
why ip_channel is used here? You can use dstDescPtr->c and remove the branch statement